### ISSUE TITLE
panic: block.AppHash does not match AppHash after replay

### ISSUE THREAD

**Issue by vega-paul** [vegaprotocol/vega#5495]
Pulled all repos and started VC network locally

TM panics

panic: block.AppHash does not match AppHash after replay. Got EAF5BC1C378D514016BDDB79FC26DC9573573785C68AF7B652E05178FFE3C937, expected 280CBBE32A572177835BF2DB83A41BFB62C11BA8DA0879207EE8B3CB6F83BB2C.

```
goroutine 1 [running]:
github.com/tendermint/tendermint/consensus.assertAppHashEqualsOneFromBlock({0xc002014a00, 0x20, 0x20}, 0xc0018ad4a0)
	/Users/paulwebb/repos/vega/vendor/github.com/tendermint/tendermint/consensus/replay.go:515 +0x10d
github.com/tendermint/tendermint/consensus.(*Handshaker).replayBlocks(_, {{{0xb, 0x1}, {0xc0014571f8, 0x7}}, {0xc000574150, 0x11}, 0x1, 0x5a9e, {{0xc000012c60, ...}, ...}, ...}, ...)
	/Users/paulwebb/repos/vega/vendor/github.com/tendermint/tendermint/consensus/replay.go:468 +0x1fd
github.com/tendermint/tendermint/consensus.(*Handshaker).ReplayBlocks(_, {{{0xb, 0x1}, {0xc0014571f8, 0x7}}, {0xc000574150, 0x11}, 0x1, 0x5a9e, {{0xc000012c60, ...}, ...}, ...}, ...)
	/Users/paulwebb/repos/vega/vendor/github.com/tendermint/tendermint/consensus/replay.go:408 +0x8bc
github.com/tendermint/tendermint/consensus.(*Handshaker).Handshake(0xc003e58aa8, {0x29d1748, 0xc0013e0820})
	/Users/paulwebb/repos/vega/vendor/github.com/tendermint/tendermint/consensus/replay.go:268 +0x3c8
github.com/tendermint/tendermint/node.doHandshake({_, _}, {{{0xb, 0x1}, {0xc0014571f8, 0x7}}, {0xc000574150, 0x11}, 0x1, 0x5a9e, ...}, ...)
	/Users/paulwebb/repos/vega/vendor/github.com/tendermint/tendermint/node/node.go:325 +0x1b8
github.com/tendermint/tendermint/node.NewNode(0xc00137a140, {0x29c1bd8, 0xc001325900}, 0xc001464f40, {0x29b5cc0, 0xc001407b60}, 0xe8?, 0xc001456578?, 0xc0014650a0, {0x29c5f30, ...}, ...)
	/Users/paulwebb/repos/vega/vendor/github.com/tendermint/tendermint/node/node.go:733 +0x577
main.defaultNewNode(0xc00137a140, {0x29c5f30, 0xc001433a40})
	/Users/paulwebb/repos/vega/cmd/vega/tm_node.go:36 +0x3dc
main.newRunNodeCmd.func1(0xc000177b80?, {0xc00017fae0?, 0x0?, 0x2?})
	/Users/paulwebb/repos/vega/cmd/vega/tm_node.go:145 +0x2b5
github.com/spf13/cobra.(*Command).execute(0xc000177b80, {0xc00017fac0, 0x2, 0x2})
	/Users/paulwebb/repos/vega/vendor/github.com/spf13/cobra/command.go:856 +0x67c
github.com/spf13/cobra.(*Command).ExecuteC(0x35ac740)
	/Users/paulwebb/repos/vega/vendor/github.com/spf13/cobra/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/paulwebb/repos/vega/vendor/github.com/spf13/cobra/command.go:902
github.com/tendermint/tendermint/libs/cli.Executor.Execute({0x35ac740?, 0x27abce0?})
	/Users/paulwebb/repos/vega/vendor/github.com/tendermint/tendermint/libs/cli/setup.go:89 +0x4a
main.(*tmCmd).Execute(0x35316e0?, {0x55?, 0x3a6b758?, 0x11?})
	/Users/paulwebb/repos/vega/cmd/vega/tm.go:52 +0x26a
main.Main({0x29c5b40, 0xc00003e100})
	/Users/paulwebb/repos/vega/cmd/vega/main.go:50 +0x79
main.main()
	/Users/paulwebb/repos/vega/cmd/vega/main.go:40 +0x27
```

**Comment by ze97286** [comment]
@vega-paul in the archive.zip there's only one node, where's the full log of the other node? When you open such bugs please include *ALL* logs or else there's not much we can do.

**Comment by wwestgarth** [comment]
I did look at this at the time (Paul sent me his entire block-data) the full logs indicate that the chain was replaying, so what I suspect had happened was that the replay was happening with a slightly different version of vega, so a snapshot was slightly different.

I replayed a chain locally against the block-data and consistently got the "wrong" hash across all nodes, and couldn't replicate any non-determinisim. So I don't think there is any issue here other than not clearing state between network restarts.
